File: android/app/src/main/java/com/openburnbar/data/policy/MobileProductSurfacePolicy.kt
package com.openburnbar.data.policy

enum class MobileProductCardDisposition(val wire: String) {
    REAL("real"),
    REMOVED("removed"),
    GATED("gated"),
}

/** Interactive card actions must be real, gated, or removed — never decorative. */
object MobileProductSurfacePolicy {
    fun disposition(
        actionId: String,
        catalogPresent: Boolean = true,
        entitlement: MobileStoreEntitlementState = MobileStoreEntitlementState.NONE,
    ): MobileProductCardDisposition = when (actionId) {
        "decorative.stop", "fake.cancel", "silent.discard" -> MobileProductCardDisposition.REMOVED
        "store.purchase", "store.restore" ->
            if (catalogPresent) MobileProductCardDisposition.REAL else MobileProductCardDisposition.REMOVED
        "budget.enforce", "surface.budget-enforce" ->
            if (entitlement == MobileStoreEntitlementState.ACTIVE) {
                MobileProductCardDisposition.REAL
            } else {
                MobileProductCardDisposition.GATED
            }
        "inbox.archive", "inbox.snooze", "inbox.feedback", "inbox.open-route",
        "store.open", "pulse.retry", "streams.retry",
        -> MobileProductCardDisposition.REAL
        else -> MobileProductCardDisposition.REMOVED
    }

    fun mayEnforceBudget(entitlement: MobileStoreEntitlementState): Boolean =
        entitlement == MobileStoreEntitlementState.ACTIVE
}

File: OpenBurnBarCore/Sources/OpenBurnBarMobilePolicy/MobileProductSurfacePolicy.swift
import Foundation

public enum MobileProductCardDisposition: String, Sendable, Equatable {
    case real
    case removed
    case gated
}

/// Interactive card actions must be real, gated, or removed — never decorative.
public enum MobileProductSurfacePolicy {
    public static func disposition(
        actionId: String,
        catalogPresent: Bool = true,
        entitlement: MobileStoreEntitlementState = .none
    ) -> MobileProductCardDisposition {
        switch actionId {
        case "decorative.stop", "fake.cancel", "silent.discard":
            return .removed
        case "store.purchase", "store.restore":
            if !catalogPresent { return .removed }
            return .real
        case "budget.enforce", "surface.budget-enforce":
            return entitlement == .active ? .real : .gated
        case "inbox.archive", "inbox.snooze", "inbox.feedback", "inbox.open-route",
             "store.open", "pulse.retry", "streams.retry":
            return .real
        default:
            return .removed
        }
    }

    public static func mayEnforceBudget(_ entitlement: MobileStoreEntitlementState) -> Bool {
        entitlement == .active
    }
}
